OpenRA.Game/Map/MapCache.cs

```diff
@@ -159,50 +159,57 @@ public IEnumerable<Map> EnumerateMapsWithoutCaching(MapClassification classifica
 				yield return new Map(modData, mapPackage);
 		}
 
-		public void QueryRemoteMapDetails(string repositoryUrl, IEnumerable<string> uids, Action<MapPreview> mapDetailsReceived = null, Action queryFailed = null)
+		public void QueryRemoteMapDetails(string repositoryUrl, IEnumerable<string> uids, Action<MapPreview> mapDetailsReceived = null, Action<MapPreview> mapQueryFailed = null)
 		{
-			var maps = uids.Distinct()
+			var queryUids = uids.Distinct()
 				.Where(uid => uid != null)
 				.Select(uid => previews[uid])
 				.Where(p => p.Status == MapStatus.Unavailable)
-				.ToDictionary(p => p.Uid, p => p);
+				.Select(p => p.Uid)
+				.ToList();
 
-			if (!maps.Any())
-				return;
-
-			foreach (var p in maps.Values)
-				p.UpdateRemoteSearch(MapStatus.Searching, null);
-
-			var url = repositoryUrl + "hash/" + string.Join(",", maps.Keys) + "/yaml";
-
-			Task.Run(async () =>
+			// Limit each query to 50 maps at a time to avoid request size limits
+			for (var i = 0; i < queryUids.Count; i += 50)
```

Comment:
Shouldnt this loop be inside the task and it can use the same httplclient?

Reply:
wouldn't this then force the queries to run serially instead of in parallel?

Maybe this is would be a good thing considering the shaky state of the RC backend...